README.md

```diff
@@ -104,6 +104,166 @@ TODO: specify the encoding (byte-array to string) procedure
 
 TODO: specify the decoding (string to byte-array) procedure
 
+### Codecs
+
+Depending on the protocol of a Multiaddr item, different algorithms are used to
+convert their values from/to binary representation. The name of the codec to use
+for each protocol is noted in [protocols.csv](protocols.csv).
+
+In general empty values in the string representation are always disallowed unless
+explicitely noted otherwise. In case of conversion errors implementation must
+refuse to process the given string/binary value and report the error to the caller
+instead.
+
+While all codec types may be encoded using the standard variable length encoding,
+some may also allow being encoded into a specific static-length binary value
+without the extra length information if this is noted in the respective codec's
+description.
```

Comment:
This isn't true in go (and probably not in other languages either). Only variable length codecs can use length prefixes.

Reply:
It'd be a nice transitional step for making #70 possible for unknown binary formats. But I see that this is not the right place for discussing this. I'll move this to a different PR, so that we can merge this one…